typescript_typeorm_graphql_project/src/controllers/feed.ts

```diff
@@ -0,0 +1,211 @@
+            name: existingUser.name,
+          },
+        },
+
+        message: "New Post is Created succesfully.",
+      });
+    })
+    .catch((err) => {
+      if (!err.statusCode) {
+        err.statusCode = 500;
+      }
+      next(err);
+    });
+};
+
+const getAllPost = async (
+  req: Request | any,
+  res: Response,
+  next: NextFunction
+) => {
+  Post.find().then((posts) => {
+    res
+      .status(200)
+      .json({ message: "Fetched posts successfully.", posts: posts });
+  });
+};
+
+const getPostById = async (
+  req: Request | any,
+  res: Response,
+  next: NextFunction
+) => {
+  try {
+    const postId = req.params.postId;
+    Post.findOne({
+      where: {
+        id: postId,
+      },
+    })
+      .then((post) => {
+        if (!post) {
+          res.status(404);
+          const error = new Error("Could not find post.");
+          throw error; // after throwing the error the catch block will executed.
+        }
+        res.status(200).json({ message: "Post fetched.", post: post });
+      })
+      .catch((err) => {
+        if (!err.statusCode) {
+          err.statusCode = 500;
+        }
+        next(err);
+      });
+  } catch (err) {
+    console.log(err);
+    throw err;
+  }
+};
+
+const updatePost = async (
+  req: Request | any,
+  res: Response,
+  next: NextFunction
+) => {
+  const postId = req.params.postId;
+  const errors = validationResult(req);
+
+  if (!errors.isEmpty()) {
+    res.status(422);
+    const error = new Error("Validation failed, entered data is incorrect.");
+    throw error;
+  }
+
+  const title = req.body.title;
+  const content = req.body.content;
+  const imageUrl = req.body.image;
+
+  Post.findOne({
+    where: {
+      id: postId,
+    },
+    relations: {
+      creator: true,
+    },
+  })
+    .then((post) => {
+      if (!post) {
+        res.status(404);
+        const error = new Error("Could not find post.");
+        throw error;
+      }
+
+      if (post.creator.id !== parseInt(req.userId)) {
+        res.status(403);
+        const error = new Error("Not Authorized.");
+        throw error;
+      }
+      post.title = title;
+      post.imageUrl = imageUrl;
+      post.content = content;
+      return post.save();
+    })
+    .then((result) => {
+      res.status(200).json({ message: "Post updated!", post: result });
+    })
+    .catch((err) => {
+      if (!err.statusCode) {
+        err.statusCode = 500;
+      }
+      next(err);
+    });
+};
+
+const deletePostById = (
+  req: Request | any,
+  res: Response,
+  next: NextFunction
+) => {
+  const postId = req.params.postId;
+
+  Post.findOne({
+    where: {
+      id: postId,
+    },
+    relations: {
+      creator: true,
+    },
+  })
+    .then((post) => {
+      if (!post) {
+        res.status(404);
+        const error = new Error("Could not fing post.");
+        throw error;
+      }
+
+      if (post.creator.id !== parseInt(req.userId)) {
+        res.status(403);
```

Comment:
statusCode 403 is required for Forbidden, not for unauthorized
for unauthorized you can use 401